compiler/optimizer/LoopVersioner.hpp

```diff
@@ -229,7 +229,7 @@ class TR_LoopVersioner : public TR_LoopTransformer
    void buildDivCheckComparisonsTree(List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::Node> *, TR::Block *);
    void buildIwrtbarComparisonsTree(List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::Node> *, TR::Block *);
    void buildCheckCastComparisonsTree(List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::Node> *, TR::Block *);
-   void buildConditionalTree(List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::Node> *, TR::Block *, SharedSparseBitVector &reverseBranchInLoops);
+   void buildConditionalTree(TR_ScratchList<TR::Block> *blocksInWhileLoop, List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::TreeTop> *, List<TR::Node> *, TR::Block *, SharedSparseBitVector &reverseBranchInLoops);
```

Comment:
I wonder if it's necessary to add the extra argument/logic to the buildConditionalTree routine since the detectInvariantTrees routine that runs earlier in the flow logic of the loop versioner would have already filtered out those cases when the guard target is not inside the loop (courtesy of your change in this commit to  detectInvariantTrees)